src/Provider/PackageArchiver.php

```diff
@@ -197,7 +197,7 @@ protected function archive_package( string $slug, string $type ): Package {
 		try {
 			$package = $this->packages->first_where( compact( 'slug', 'type' ) );
 
-			if ( null !== $package ) {
+			if ( null !== $package && $package->is_installed() ) {
```

Comment:
Logically at run time, this should work.

But static analysis will still show that `$package` can be any type of `Package`, including one that hasn't got an `get_installed_release()`. i.e. we're using the human knowledge of having to remember to check for `is_installed()` before relying on `get_installed_release()`. This is something that an interface should be providing.

![screenshot 2018-07-11 13 33 38](https://user-images.githubusercontent.com/88371/42571470-26762ea2-850f-11e8-919e-a4220763a9c5.png)